File: TI1.1 SOAP_JAVA_SINBDD_GR06/01 SERVIDOR/ConUni_Soap_Java_GR06/src/java/ec/edu/monster/modelo/AutenticacionService.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.modelo;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author josue
 */
public class AutenticacionService {
    private static List<Usuario> usuariosRegistrados;
    
    // Bloque estático para inicializar usuarios quemados
    static {
        usuariosRegistrados = new ArrayList<>();
        usuariosRegistrados.add(new Usuario("MONSTER", "MONSTER9"));
    }
    
    /**
     * Autentica un usuario con username y password
     * @param username nombre de usuario
     * @param password contraseña
     * @return Usuario si las credenciales son correctas, null si no
     */
    public static Usuario autenticar(String username, String password) {
        for (Usuario usuario : usuariosRegistrados) {
            if (usuario.getUsername().equals(username) && 
                usuario.getPassword().equals(password)) {
                return usuario;
            }
        }
        return null;
    }
    
    /**
     * Verifica si un usuario existe
     * @param username nombre de usuario
     * @return true si existe, false si no
     */
    public static boolean existeUsuario(String username) {
        for (Usuario usuario : usuariosRegistrados) {
            if (usuario.getUsername().equals(username)) {
                return true;
            }
        }
        return false;
    }
    
    /**
     * Obtiene todos los usuarios (sin contraseñas)
     * @return lista de usuarios
     */
    public static List<Usuario> obtenerTodosLosUsuarios() {
        return new ArrayList<>(usuariosRegistrados);
    }
}

File: TI1.4 RESTFUL_DOTNET_SINBDD_GR06/ConUni_RestFul_Dotnet_GR06/ConUni_RestFul_Dotnet_GR06/Servicios/AutenticacionService.cs
using ConUni_RestFul_Dotnet_GR06.Modelos;

namespace ConUni_RestFul_Dotnet_GR06.Servicios
{
    /// <summary>
    /// Servicio de autenticación con usuarios hardcodeados
    /// </summary>
    public class AutenticacionService
    {
        private static readonly List<Usuario> UsuariosRegistrados;

        static AutenticacionService()
        {
            UsuariosRegistrados = new List<Usuario>
            {
                new Usuario("MONSTER", "MONSTER9")
            };
        }

        /// <summary>
        /// Autentica un usuario con username y password
        /// </summary>
        public static Usuario Autenticar(string username, string password)
        {
            return UsuariosRegistrados.FirstOrDefault(u =>
                u.Username.Equals(username, StringComparison.Ordinal) &&
                u.Password.Equals(password, StringComparison.Ordinal));
        }

        /// <summary>
        /// Verifica si un usuario existe
        /// </summary>
        public static bool ExisteUsuario(string username)
        {
            return UsuariosRegistrados.Any(u =>
                u.Username.Equals(username, StringComparison.Ordinal));
        }

        /// <summary>
        /// Obtiene todos los usuarios (sin contraseñas)
        /// </summary>
        public static List<Usuario> ObtenerTodosLosUsuarios()
        {
            return new List<Usuario>(UsuariosRegistrados);
        }
    }
}
